From 5ced733966ed0fc7ebaf876ff61bbe6af6d11b1b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 02 Jun 2019 11:05:16 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: let reconfiguration

The CPU and PCI PLLs are configured in BTL in a "gated" manner to W/A
the PLL clock switch glitch. Therefore these PLLs should not be
reconfigured in preboot.
All other PLLs can be reconfigured, so let their (re)configuration.

Change-Id: Ifeb4a9b904a05afdf31ec41bf62bb412330a97d9
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 3887b68..13980ad 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -175,21 +175,9 @@
 
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
 {
-	enum hl_pll_bypass_val bval;
-
 	switch (cfg->base) {
-	case mmMESH_W_PLL_BASE:
-	case mmMESH_E_PLL_BASE:
-	case mmIF_W_PLL_BASE:
-	case mmIF_E_PLL_BASE:
 	case mmPSOC_PCI_PLL_BASE:
-		/* DIV0 is the undivided PLL output */
-		bval = hl_pll_bypass_get(cfg->base, HL_PLL_DIV0);
-		if (bval == HL_PLL_PLL_CLK || bval == HL_PLL_DIV_PLL)
-			return false;
-		break;
 	case mmPSOC_CPU_PLL_BASE:
-		/* No CPU PLL configuration while running on it */
 		return false;
 	default:
 		break;
